### DISCUSSION TITLE
http.route set to {{contextPath}}/* for all request not mapped in servletMapping

### DISCUSSION THREAD

**Discussion by aditya-32** [open-telemetry/opentelemetry-java-instrumentation#10444] [Q&A] [answered]
In Otel for all requests hit to a path that is not exposed or mapped in servletMapping (servletRegistration.getMappings()), http.route is set to {servletContextPath}/*
for example 
i'm running a java application  using tomcat 10 with jakarta servlets and i have mapped **_/rtb/internal/openx_** in servletMappings and if i make request at  /**_rtb/internal/openx_** then the http.route is correct and set to value **_/rtb/internal/openx_** but if i make request to **_/rtb/internal/openx/_** then since it was not exposed not added to servletMappings, hence it's http.route is set to **_/rtb/*_**. 

Can someone please help me in understanding the use case as well as where can i find the code in [repo](https://github.com/open-telemetry/opentelemetry-java-instrumentation) , i tried checking for it in tomcat as well as servlet instrumentation but could not find exact code which is responsible for this transformation/updation for http.route

**Comment by laurit** [comment, marked as answer]
See https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/1c07aa5c954caa4b47f39e20b1ec6e1f84b7879d/instrumentation/servlet/servlet-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/ServletSpanNameProvider.java#L25

**Comment by aditya-32** [comment]
is there a way to het the actual path for those 404 cases as well?

**Reply by laurit** [reply]
https://opentelemetry.io/docs/specs/semconv/http/http-spans/#name states that the span name must have a low cardinality. Paths that return 404 can potentially have a high cardinality. Spec says

> Instrumentation MUST NOT default to using URI path as span name, but MAY provide hooks to allow custom logic to override the default span name.

It is possible to use manual instrumentation to overwrite the span name set by the agent.
